### DISCUSSION TITLE
At what point Mods are being applied?

### DISCUSSION THREAD

**Discussion by Regturrr** [Romanitho/Winget-AutoUpdate#481] [General]
Hey guys,

First of all, what a great tool!

I've set up a test environment which syncs mods from an Azureblob location succesfully.
When I manually run the 'WAU - Check for updated Apps' shortcut, it syncs the mods and places them in the correct local folder.

But from that moment it's being quiet.. I've installed the Zoom application (ID: Zoom.Zoom) and created a mod file named 'Zoom.Zoom-installed.ps1' with the following content:

```
<# FUNCTIONS #>
. $PSScriptRoot\_Mods-Functions.ps1

<# ARRAYS/VARIABLES #>

#Beginning of Desktop Link Name to Remove - optional wildcard (*) after, without .lnk, multiple: "lnk1","lnk2"
$Lnk = @("Zoom")

<# MAIN #>

if ($Lnk) {
    Remove-ModsLnk $Lnk
}
```

I want to remove the desktop shortcut, but not seem able to do so. The only way I get it working is to place the file '_WAU_Mods.ps1' with the following content in the Azureblob location:

```
<# #Mods for WAU (if Network is active/any Winget is installed/running as SYSTEM)
Winget-Upgrade.ps1 calls this script with the code:
[Write-ToLog "Running Mods for WAU..." "Yellow"
& "$Mods\_WAU-mods.ps1"]
Make sure your Functions have unique names!
Exit 1 to Re-run WAU from this script!
#>

<# FUNCTIONS #>
. $PSScriptRoot\_Mods-Functions.ps1

<# ARRAYS/VARIABLES #>

#Beginning of Desktop Link Name to Remove - optional wildcard (*) after, without .lnk, multiple: "lnk1","lnk2"
$Lnk = @("Zoom")

<# MAIN #>

if ($Lnk) {
    Remove-ModsLnk $Lnk
}

Write-ToLog "...everything's already been done!" "Green"
Exit 0
```

This file is also synced correctly and when running the 'WAU - Check for updated Apps' shortcut, it deletes the shortcut from the desktop.

What am I doing wrong?

**Comment by KnifMelti** [comment]
In the above written I spot an error, it should be:
`. $PSScriptRoot\_Mods-Functions.ps1`

**Reply by KnifMelti** [reply]
Ah, site changed it...

**Reply by KnifMelti** [reply]
The -installed mod runs when the app is upgraded successfully next time WAU runs, not if it's already installed and not upgrading via WAU

**Reply by Regturrr** [reply]
Do I need to use a 'Zoom.Zoom-install.ps1' file instead? I want to delete the desktop shortcut immediately / asap after I install the app. Is there any other way to apply a mod after the initial install of an app besides the _WAU_Mods.ps1?

**Reply by KnifMelti** [reply]
https://github.com/Romanitho/Winget-AutoUpdate#custom-scripts-mods-feature-for-apps
Read about when the different Mods are running

If you install the app with Winget-Install and use a Mod for it, yes
https://github.com/Romanitho/Winget-Install#custom-mods

**Reply by Regturrr** [reply]
Thanks for your answer.

I've tried to delete the shortcut using the following files:

- Zoom.Zoom-installed.ps1
- Zoom.Zoom-install.ps1
- Zoom.Zoom-install-once.ps1

The file contains the following:

<# ARRAYS/VARIABLES #>

<# FUNCTIONS #>

<# MAIN #>
$FileName = "C:\Users\Public\Desktop\Zoom.lnk"
if (Test-Path $FileName) {
   Remove-Item $FileName -verbose
}

<# EXTRAS #>

When I manually run the file, the shortcut is deleted correctly.

When I use the file called '_WAU_Mods.ps1' it's executed correctly when I use the 'Check for updates Apps' shortcut.

What am I doing wrong here?

**Reply by KnifMelti** [reply]
### _A script Template and Mods Functions are included as an **example** to get you started..._

I don't see that your calling the:
```
<# FUNCTIONS #>
. $PSScriptRoot\_Mods-Functions.ps1
```

**I propose that you' write your own script all together** with the things you want to do when **Zoom** gets installed/updated in one (1) `Zoom.Zoom-install.ps1` and place it under a `mods` folder where your `winget-install.ps1` is located, eg:

```
mods
  Zoom.Zoom-install.ps1

winget-install.ps1
```
And then run:
`powershell.exe -Executionpolicy Bypass -File [PATH_TO]winget-install.ps1 -AppIDs Zoom.Zoom`

**Reply by Regturrr** [reply]
Thanks for your quick reply. I've tried both with and without the use of the functions.
I've just downloaded the latest 'winget-install.ps1' and that seemed to be the problem. The version I was using was way different than this one.

The shortcut is not instantly removed from the desktop after installing the application.

Thanks for the support :)
